eodag/api/product/_assets.py

```diff
@@ -54,6 +57,31 @@ def as_dict(self) -> Dict[str, Any]:
         """
         return {k: v.as_dict() for k, v in self.data.items()}
 
+    def get_values(self, asset_filter: str = "") -> List[Asset]:
+        """
+        retrieves the assets matching the given filter
+        :param asset_filter: filter with which the assets should be matched
+        :type asset_filter: str
+        :return: list of assets
+        :rtype: List[Asset]
+        """
+        if asset_filter:
+            filter_regex = re.compile(asset_filter)
+            assets_keys = self.keys()
+            assets_keys = list(filter(filter_regex.fullmatch, assets_keys))
```

Comment:
Incompatible types in assignment (expression has type "list[Any]", variable has type "KeysView[Any]")